### PR TITLE
Aks migration changes

### PR DESCRIPTION
**Pull request by adusumillipraveen**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-package-lock.json
+!package.json
+!yarn.lock
+!app/**
+!config/**
+!locale/**
+!public/**
+!views/**
+
+!nodemon.js
```
Comment (timja) [inline .dockerignore]: shouldn't need this

File: .dockerignore [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+!tsconfig.json
+!tslint.json
```
Comment (timja) [inline .dockerignore]: would be surprised if you need this

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -35,14 +35,15 @@ static LinkedHashMap<String, Object> secret(String secretName, String envVar) {
   ]
 }
 
+def yarnBuilder = new uk.gov.hmcts.contino.YarnBuilder(this)
+
 withPipeline("nodejs", product, component) {
     loadVaultSecrets(secrets)
```
Comment (timja) [inline Jenkinsfile_CNP]: are there any secrets that are no longer needed anymore? I wouldn't be surprised if none are needed

File: app/server/server.ts [hunk 4]
```diff
@@ -1,7 +1,12 @@
+const config = require('@hmcts/properties-volume').addTo(require('config'));
```
Comment (timja) [inline app/server/server.ts]: did the import code not work?

File: app/server/server.ts [hunk 5]
```diff
@@ -1,7 +1,12 @@
...
+const setupSecrets = require('./services/setupSecrets');
```
Comment (timja) [inline app/server/server.ts]: can't we import this or at least import just the single function that is there?

```suggestion
import { setupKeyVaultSecrets } from './services/setupSecrets';
```

alternatively we could change the function to be the default export and then do:
```suggestion
import setupKeyVaultSecrets from './services/setupSecrets';
```

File: app/server/server.ts [hunk 6]
```diff
@@ -1,7 +1,12 @@
...
+
+// Setup secrets before loading the app
+setupSecrets.setupKeyVaultSecrets();
```
Comment (timja) [inline app/server/server.ts]: if you import the function this can be changed like so:
```suggestion
setupKeyVaultSecrets();
```

File: app/server/services/setupSecrets.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import * as config from 'config';
+import * as propertiesVolume from '@hmcts/properties-volume';
+import { get, set } from 'lodash';
+
+propertiesVolume.addTo(config);
```
Comment (timja) [inline app/server/services/setupSecrets.ts]: why is this added here and in `server.ts`?

File: app/server/services/setupSecrets.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+function setSecret (secretPath: string, configPath: string) {
```
Comment (timja) [inline app/server/services/setupSecrets.ts]: normally you should always specify a type for the return value
```suggestion
function setSecret (secretPath: string, configPath: string): void {
```

File: app/server/services/setupSecrets.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  // Only overwrite the value if the secretPath is defined
+  if (config.has(secretPath)) {
+    set(config, configPath, get(config, secretPath));
+  }
+}
+
+export function setupKeyVaultSecrets () {
```
Comment (timja) [inline app/server/services/setupSecrets.ts]:
```suggestion
export function setupKeyVaultSecrets (): void {
```

could also make it the default function as I don't expect you to need to export anything else from here,

```suggestion
export default function setupKeyVaultSecrets () {
```

File: charts/sscs-cor-frontend/values.preview.template.yaml [hunk 10]
```diff
@@ -1,23 +1,23 @@
...
   environment:
-    SSCS_API_URL: "http://sscs-cor-backend-aat.service.core-compute-aat.internal"
-    IDAM_URL: "https://idam-web-public.aat.platform.hmcts.net"
-    IDAM_API_URL: "https://idam-api.aat.platform.hmcts.net"
-    TRIBUNALS_API_URL: "http://sscs-tribunals-api-aat.service.core-compute-aat.internal"
-    IDAM_CLIENT_SECRET: ${IDAM_CLIENT_SECRET}
-    S2S_URL: "http://rpe-service-auth-provider-aat.service.core-compute-aat.internal"
     REDIS_URL: redis://${SERVICE_NAME}-redis-master
     SESSION_SECRET:
     NODE_ENV: preview
```
Comment (timja) [inline charts/sscs-cor-frontend/values.preview.template.yaml]: can probably bin this?

File: charts/sscs-cor-frontend/values.preview.template.yaml [hunk 11]
```diff
@@ -1,23 +1,23 @@
...
+    EVIDENCE_UPLOAD_QUESTION_PAGE_OVERRIDE_ALLOWED: "true"
+  keyVaults:
+    sscs:
+      secrets:
+        - sscs-cor-idam-client-secret
+        - sscs-s2s-secret
+redis:
+  enabled: true
+tags:
+  - idam-pr
```
Comment (timja) [inline charts/sscs-cor-frontend/values.preview.template.yaml]: is it better to change this to a condition for consistency?

File: .dockerignore (hunk 1 above)
Comment (adusumillipraveen) [inline .dockerignore]:
```suggestion

```

File: .dockerignore (hunk 2 above)
Comment (adusumillipraveen) [inline .dockerignore]:
```suggestion
```

File: Jenkinsfile_CNP (hunk 3 above)
Comment (adusumillipraveen) [reply Jenkinsfile_CNP]: Yes, they are needed. Deleted them and added one by one functional tests failed

File: app/server/services/setupSecrets.ts [hunk 12]
```diff
@@ -2,16 +2,14 @@ import * as config from 'config';
 import * as propertiesVolume from '@hmcts/properties-volume';
```
Comment (timja) [inline app/server/services/setupSecrets.ts]: is this still needed?

File: charts/sscs-cor-frontend/values.preview.template.yaml (hunk 10 above)
Comment (adusumillipraveen) [reply charts/sscs-cor-frontend/values.preview.template.yaml]: Had some references in code, thought i better leave it there.

File: test/unit/services/setupSecrets.test.ts [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+          'sscs-cor-redis-access-key': 'redisValue',
+          'sscs-cor-idam-client-secret': 'idamValue',
+          'sscs-s2s-secret': 'osPlacesValue'
+        }
+      };
+
+      // Update config with secret setup
+      const setupSecrets = proxyquire(modulePath,
+        { config: mockConfig });
+      setupSecrets.setupKeyVaultSecrets();
```
Comment (timja) [inline test/unit/services/setupSecrets.test.ts]: I think this will fix it
```suggestion
      setupSecrets();
```
